parsers/SG.py

```diff
@@ -267,6 +232,79 @@ def fetch_price(zone_key='SG', session=None, target_datetime=None, logger=None)
     }
 
 
+def __detect_datetime_from_solar_image(solar_image, logger):
+    w, h = solar_image.size
+    crop_left = int(w * 0.75)
+    crop_top = int(h * 0.87)
+    crop_right = int(w * 0.93)
+    crop_bottom = int(h * 0.92)
+    time_img = solar_image.crop((crop_left, crop_top, crop_right, crop_bottom))
+    processed_img = __preprocess_image_for_ocr(time_img)
```

Comment:
Result after new cropping and pre-processing technique is:
![pv-datetime-preprocess-result](https://user-images.githubusercontent.com/3433964/147286919-2d54db3d-2237-464b-a2c8-52651f7cd4c1.png)